### PR TITLE
Add tables and models for rate engine domestic pricing

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR adds the tables and models for representing domestic pricing in the rate engine.

## Setup

- `make db_dev_migrate` should run the migration for the new tables.
- `make server_test` should run the validation tests on the new Pop models.
- Verify that the new models seem to match the table structure.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168357735) for this change
* See this [document](https://docs.google.com/document/d/1yLU-QBxuRTmZSzhZAm4LqAqxAnESVIxMGGeR9xHRZlM/edit#heading=h.8evdt4y2wpbo) for writeup of the domestic pricing data model.
* See the [pricing template](https://docs.google.com/spreadsheets/d/1j2kMIykyUQf8Dy8NF_L4PpU9iscQCbenmqF3T5-Dy6k/edit#gid=1448507886) that inspired this design.

### CODE REVIEW

File: pkg/models/re_domestic_other_price.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	ID           uuid.UUID  `json:"id" db:"id"`
+	ContractID   uuid.UUID  `json:"contract_id" db:"contract_id"`
+	ServiceID    uuid.UUID  `json:"service_id" db:"service_id"`
+	IsPeakPeriod bool       `json:"is_peak_period" db:"is_peak_period"`
+	Schedule     int        `json:"schedule" db:"schedule"`
+	PriceCents   unit.Cents `json:"price_cents" db:"price_cents"`
+	CreatedAt    time.Time  `json:"created_at" db:"created_at"`
+	UpdatedAt    time.Time  `json:"updated_at" db:"updated_at"`
+
+	// Associations
```
Comment (lynzt) [inline pkg/models/re_domestic_other_price.go]: Just curious if we're listing associations last... I've seen other models where the Association is right after the associated ID...

Comment (reggieriser) [reply pkg/models/re_domestic_other_price.go]: I've seen it both ways -- pretty sure the old shipment model grouped them.  I tend to lean toward grouping them together, but I'm not particularly tied to it.

Comment (lynzt) [reply pkg/models/re_domestic_other_price.go]: just curious and works for me

**Comment by reggieriser** [comment]
> Question: I assume `re` is for `rate engine`? Is there a reason we're prefixing these w/ `re`?
> To me, it seems redundant...

Regarding table names, we modeled them after the way we grouped the `tariff400ng_*` tables of imported data for the previous rate engine.  The associated Pop models were also prefixed by `Tariff400ng` as well.  Pop assumes the table name for its queries based on the model name, but I believe we can add a `TableName()` function to the model and get around this if we want.

I'll discuss with the others working on these models (we have separate stories that cover the whole new RE set) and put in another story to adjust them as a group if we think it makes sense.

**Comment by lynzt** [comment]
@reggieriser that makes sense to me... and it is nice they "filter" together... 🚢

**Pull request merged**